exasol/error/_cli.py

```diff
@@ -0,0 +1,354 @@
+    error_definitions = list()
+    warnings = list()
+    for f in args.python_file:
+        collector = ErrorCollector(f.name)
+        root_node = ast.parse(f.read())
+        for n in ast.walk(root_node):
+            collector.visit(n)
+
+        errors = collector.errors
+        if errors:
+            print("\n".join(str(e) for e in errors), file=sys.stderr)
+            return ExitCode.FAILURE
+
+        error_definitions.extend(collector.error_definitions)
+        warnings.extend(collector.warnings)
+
+    for w in warnings:
+        print(w, file=sys.stderr)
+
+    for e in error_definitions:
+        print(json.dumps(e, cls=_JsonEncoder))
+
+    return ExitCode.SUCCESS
+
+
+def generate_command(args) -> ExitCode:
+    """Generate an error code file for the specified workspace
+
+
+    Improvements, future TODO's (see issue#<to be created>):
+
+        Add support for:
+        * Derive python files from pyproject.toml package definition
+        * Derive name form pyproject.toml
+        * Derive version from pyproject.toml
+    """
+
+    def _report(project_name, project_version, errors):
+        return {
+            "$schema": "https://schemas.exasol.com/error_code_report-1.0.0.json",
+            "projectName": project_name,
+            "projectVersion": project_version,
+            "errorCodes": [e for e in errors],
+        }
+
+    error_definitions = list()
+    warnings = list()
+    paths = [Path(p) for p in args.root]
+    files = {f for f in chain.from_iterable([root.glob("**/*.py") for root in paths])}
+    for file in files:
+        with open(file, "r") as f:
+            collector = ErrorCollector(f.name)
+            root_node = ast.parse(f.read())
+            for n in ast.walk(root_node):
+                collector.visit(n)
+
+            errors = collector.errors
+            if errors:
+                print("\n".join(str(e) for e in errors), file=sys.stderr)
+                return ExitCode.FAILURE
+
+            error_definitions.extend(collector.error_definitions)
+            warnings.extend(collector.warnings)
+
+    for w in warnings:
+        print(w, file=sys.stderr)
+
+    error_catalogue = _report(args.name, args.version, error_definitions)
+    print(json.dumps(error_catalogue, cls=_JsonEncoder))
+
+    return ExitCode.SUCCESS
+
+
+def _create_parser():
+    parser = argparse.ArgumentParser(
+        prog="ec",
+        description="Error Crawler",
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+    )
+    subparsers = parser.add_subparsers()
+
+    parse = subparsers.add_parser(
+        name="parse",
+        description="parses error definitions out of python files and reports them in jsonl format",
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+    )
+    parse.add_argument(
+        "python_file",
+        metavar="python-file",
+        type=argparse.FileType("r"),
+        default=[sys.stdin],
```

Comment:
does this mean, the input comes from stdin?

Reply:
TL;DR: Yes

If nothing is specified it reads stdin, so commands like:

```
python -m exasol.error parse < some-python-file.py
cat some-pytho-file.py | python -m exasol.error parse
```

will work just fine.